packages/perspective-viewer-d3fc/src/ts/legend/filter.ts

```diff
@@ -13,11 +13,36 @@
 import { groupFromKey } from "../series/seriesKey";
 import { DataRowsWithKey, Settings } from "../types";
 
+function refineDateData(
+    settings: Settings,
+    data: any[] | undefined = undefined
+) {
+    let dataToRefine = data ?? settings.data;
+    const { crossValues } = settings;
+
+    crossValues.forEach(({ type }, index) => {
+        const formatType =
+            type === "date" || type === "timestamp" || type === "datetime"
+                ? (value: any) => new Date(value).toLocaleDateString()
```

Comment:
We should really use the formatting logic from `perspective-viewer-datagrid` here ... but in the interim, this only works for `date` Perspective types. `datetime` types should use `toLocaleString()` for now please.